src/k8s/pkg/k8sd/app/hooks_start.go

```diff
@@ -61,6 +62,33 @@ func (a *App) onStart(ctx context.Context, s state.State) error {
 			func(ctx context.Context) (types.ClusterConfig, error) {
 				return databaseutil.GetClusterConfig(ctx, s)
 			},
+			func() (string, error) {
+				c, err := s.Leader()
+				if err != nil {
+					return "", fmt.Errorf("failed to get leader client: %w", err)
+				}
+
+				clusterMembers, err := c.GetClusterMembers(ctx)
+				if err != nil {
+					return "", fmt.Errorf("failed to get cluster members: %w", err)
+				}
+
+				// Check if any of the cluster members have an IPv6 address, if so return "::1"
+				// if one member has an IPv6 address, other members should also have IPv6 interfaces
```

Comment:
Quick question:

> other members should also have IPv6 interfaces

Are we assuming this "should"? Maybe we can also force it?
E.g. by making sure every member has a non-nil `nodeIP.To6()`?

Reply:
This is implicit, if a member is able to join with an IPv6 address this means the other members have ipv6 interfaces. If rest of the members did not have ipv6 routing available, the join operation / clustering would never work. I'm also not exactly sure on if IPv4 / IPv6 mixed clustering is supported by microcluster and dqlite. I believe this is not something for us to enforce.